flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotSharingManager.java

```diff
@@ -316,6 +331,9 @@ public boolean contains(AbstractID groupId) {
 		// true if we are currently releasing our children
 		private boolean releasingChildren;
 
+		// the total requested by all the descendants.
+		private ResourceProfile requestedResources;
```

Comment:
I think this is a little confusing. Do you mean the resource requested by tasks that are already assigned into the slot? May be use "occupied" or "allocated" to avoid ambiguity?

Reply:
I still thought that requested might be better since the resources may have been requested before the underlying slot has been allocated for real.

Reply:
I agree that "requested" sounds confusing. I would like to change this to "reserved" if there are no objections.